eZ/Publish/API/Repository/Values/Content/Search/AggregationResult/RangeAggregationResult.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\API\Repository\Values\Content\Search\AggregationResult;
+
+use Countable;
+use eZ\Publish\API\Repository\Values\Content\Query\Aggregation\Range;
+use eZ\Publish\API\Repository\Values\Content\Search\AggregationResult;
+use Iterator;
+use IteratorAggregate;
+
+final class RangeAggregationResult extends AggregationResult implements IteratorAggregate, Countable
+{
+    /** @var \eZ\Publish\API\Repository\Values\Content\Search\AggregationResult\RangeAggregationResultEntry[] */
+    private $entries;
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Search\AggregationResult\RangeAggregationResultEntry[] $entries
+     */
+    public function __construct(string $name, iterable $entries = [])
+    {
+        parent::__construct($name);
+
+        $this->entries = $entries;
+    }
+
+    public function isEmpty(): bool
+    {
+        return empty($this->entries);
+    }
+
+    /**
+     * @return \eZ\Publish\API\Repository\Values\Content\Search\AggregationResult\RangeAggregationResultEntry[]
+     */
+    public function getEntries(): iterable
+    {
+        return $this->entries;
+    }
+
+    public function getEntry(Range $key): ?RangeAggregationResultEntry
+    {
+        foreach ($this->entries as $entry) {
+            if ($entry->getKey() == $key) {
```

Comment:
We probably should use explicit casts to `string` and do a type-safe comparison.
```php
if ((string) $entry->getKey() === (string) $key) {
```
Otherwise PHP might cast those strings into `int`s to speed up comparison and give false equality results. I don't remember from the top of my head when exactly PHP does this, but I'm pretty sure it does.

Reply:
> We probably should use explicit casts to `string` and do a type-safe comparison.
> 
> ```
> if ((string) $entry->getKey() === (string) $key) {
> ```
> 
> Otherwise PHP might cast those strings into `int`s to speed up comparison and give false equality results. I don't remember from the top of my head when exactly PHP does this, but I'm pretty sure it does.

I'd say since this is VO, it should implement `equals`. But if the intention was indeed to compare strings, then yes, above is correct.